plugins/meta/flannel/flannel.go

```diff
@@ -241,6 +248,119 @@ func cmdAdd(args *skel.CmdArgs) error {
 	return delegateAdd(args.ContainerID, n.DataDir, n.Delegate)
 }
 
+func cmdAddWindows(containerID string, n *NetConf, fenv *subnetEnv) error {
+	if n.WindowsDelegate == nil {
+		n.WindowsDelegate = make(map[string]interface{})
+	} else {
+		if hasKey(n.WindowsDelegate, "type") && !isString(n.WindowsDelegate["type"]) {
+			return fmt.Errorf("'windowsDelegate' dictionary, if present, must have (string) 'type' field")
+		}
+		if hasKey(n.WindowsDelegate, "name") {
+			return fmt.Errorf("'windowsDelegate' dictionary must not have 'name' field, it'll be set by flannel")
+		}
+		if hasKey(n.WindowsDelegate, "ipam") {
+			return fmt.Errorf("'windowsDelegate' dictionary must not have 'ipam' field, it'll be set by flannel")
+		}
+	}
+
+	n.WindowsDelegate["name"] = n.Name
+
+	if !hasKey(n.WindowsDelegate, "type") {
+		n.WindowsDelegate["type"] = "wincni.exe"
+	}
+
+	updateOutboundNat(&n.WindowsDelegate, fenv)
+
+	n.WindowsDelegate["cniVersion"] = "0.2.0"
+	if n.CNIVersion != "" {
+		n.WindowsDelegate["cniVersion"] = n.CNIVersion
+	}
+
+	n.WindowsDelegate["ipam"] = map[string]interface{}{
+		"subnet": fenv.sn.String(),
+		"routes": []interface{}{
+				map[string]interface{}{
+					"GW": calcGatewayIPforWindows(fenv.sn),
+				},
+			},
+		}
+
+	return delegateAdd(containerID, n.DataDir, n.WindowsDelegate)
+}
+
+func updateOutboundNat(windowsDelegate *map[string]interface{}, fenv *subnetEnv) {
+	if !*fenv.ipmasq {
+		return
+	}
+
+	if !hasKey(*windowsDelegate, "AdditionalArgs") {
+		(*windowsDelegate)["AdditionalArgs"] = []interface{}{}
+	}
+	addlArgs := (*windowsDelegate)["AdditionalArgs"].([]interface{})
+	nwToNat := fenv.nw.String()
+	for _, policy := range addlArgs {
+		pt := policy.(map[string]interface{})
+		if hasKey(pt, "Value") {
+			policyValue := pt["Value"]
+			switch pv := policyValue.(type) {
+			case map[string]interface{}:
+				if hasKey(pv, "Type") {
+					if strings.EqualFold(pv["Type"].(string), "OutBoundNAT") {
+						if !hasKey(pv, "ExceptionList") {
+							// add the exception since there weren't any
+							pv["ExceptionList"] = []interface{}{nwToNat}
+							return
+						}
+
+						nets := pv["ExceptionList"].([]interface{})
```

Comment:
Tabs are a scarce resource 😄   
I would suggest that the `if` branch be the negative case, in order to reduce indenting and improve readability.  You can also replace the single-case `switch` statement with a simple `if`.  For example:
```go
pt := policy.(map[string]interface{})
if !hasKey(pt, "Value") {
   continue
}
pv, ok := pt["Value"].(map[string]interface{})
if !ok || !hasKey(pv, "Type") {
   continue
}
if !strings.EqualFold(pv["Type"].(string), "OutBoundNAT") {
   continue
}
// now to the real suff
if !hasKey(pv, "ExceptionList") { ...
```

Reply:
They are scarce indeed - have made the changes and saved them for future generations :-)